wfl/pipeline/utils.py

```diff
@@ -44,7 +44,10 @@ def __init__(self, sys_name, job_name, resources, job_chunksize=-100, pre_cmds=[
 
         self.sys_name = sys_name
         self.job_name = job_name
-        self.resources = resources.copy()
+        if isinstance(resources, dict):
+            self.resources = resources.copy()
+        else:
+            self.resources = resources 
```

Comment:
I don't think having different copy semantics for dicts and objects is a good idea.  Do we need to implement a copy method for the object?

Reply:
that sounds good, I'll open a new PR

Reply:
ha, of course there is `copy.deepcopy(object)`, which I think is all? `copy.copy(object)` is not enough, becaue of `Resources.partitions`.

Reply:
I see that you just handled this with a `deepcopy`.  I'm happy with that, so closing this conversation,